server/utils/secrets/secret_ref_utils.py

```
@@ -452,6 +447,47 @@ def get_user_token_data(user_id: str, provider: str) -> Optional[Dict[str, Any]]
+    cursor = None
+    try:
+        conn = connect_to_db_as_admin()
+        cursor = conn.cursor()
+        set_rls_context(cursor, conn, user_id, log_prefix="[SecretRef:tokenOwner]")
+        cursor.execute(
+            f"""SELECT user_id FROM user_tokens
+               WHERE {predicate}
+                 AND provider = %s
+                 AND secret_ref IS NOT NULL
+                 AND is_active = TRUE
+               LIMIT 1""",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)